### PR TITLE
feat:[HL-84] Filter by Main Ingredient API🍕

### PR DESCRIPTION
**Pull request by djayyab**
#### Proposed Changes
🍄 FilterByMainIngredient Entity
🍄 Add FilterByMainIngredientAPIRequest File
🍄 Add FilterByMainIngredientAPIRequestTests

#### Checklist

- [ ] I have tested these changes locally.
- [ ] I have added appropriate documentation or updated existing documentation.
- [ ] I have added appropriate test coverage or updated existing test coverage.
- [ ] I have updated the changelog (if applicable).
- [ ] I have followed the project's code style and formatting guidelines.
- [ ] I have reviewed and adhered to the project's contributing guidelines.

### CODE REVIEW

File: Networking/Sources/Networking/Requests/FilterByMainIngredientAPIRequest.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public struct FilterByMainIngredientAPIRequest: RequestType {
+    public typealias ResponseType = FilterByMainIngredientAPIResponse
+
+    public init() {}
+
+    public var baseUrl: URL { Constants.theMealDB }
+    public var path: String { "filter.php" }
+    public var method: String = "GET"
+    public var queryParameters: [String: String] {
+        ["i": "chicken_breast,garlic,salt"]
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/FilterByMainIngredientAPIRequest.swift]: These values should be injected

File: Networking/Sources/Networking/Requests/FilterByMainIngredientAPIRequest.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// MARK: - FilterByMainIngredientAPIResponse
+public struct FilterByMainIngredientAPIResponse: Codable {
+    let meals: [Meal]
+}
+
+// MARK: - Meal
+struct Meal: Codable {
+    let strMeal: String
+    let strMealThumb: String
+    let idMeal: String
```
Comment (ahmdmhasn) [inline Networking/Sources/Networking/Requests/FilterByMainIngredientAPIRequest.swift]: Please re-write the names with more descriptive names and use coding keys to match the endpoint values.

File: Healthy/Classes/Entities/FilterByMainIngredient.swift [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import Foundation
+
+// MARK: - FilterByMainIngredient
+public struct FilterByMainIngredient {
+    let meals: [Meal]
+}
+
```
Comment (ahmdmhasn) [inline Healthy/Classes/Entities/FilterByMainIngredient.swift]:
```suggestion
```

File: Healthy/Classes/Entities/FilterByMainIngredient.swift [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+// MARK: - Meal
+public struct Meal {
+    let id: String
+    let meal: String
+    let mealThumb: String
```
Comment (ahmdmhasn) [inline Healthy/Classes/Entities/FilterByMainIngredient.swift]:
```suggestion
    let thumbnailImageUrl: String
```

**Review by ahmdmhasn — APPROVED** [review]

**Review by esraakhaled — APPROVED** [review]

**Pull request merged**